domains/runtime/evm/src/lib.rs

```diff
@@ -323,14 +326,85 @@ impl pallet_balances::Config for Runtime {
     type MaxHolds = ();
 }
 
+impl pallet_domain_transaction_fees::Config for Runtime {
+    type Balance = Balance;
+}
+
+type NegativeImbalance = <Balances as Currency<AccountId>>::NegativeImbalance;
+
+/// Implementation of [`pallet_transaction_payment::OnChargeTransaction`] that charges transaction
+/// fees from the transaction sender and collect all the fee in `pallet_domain_transaction_fees`
+pub struct OnChargeTransaction;
+
+impl pallet_transaction_payment::OnChargeTransaction<Runtime> for OnChargeTransaction {
```

Comment:
Would have been nice to not implement this instead a hook to that just gives the final transaction fee. Are you sure this is not possible ?

Why not use the impl from `pallet_transaction_payment::CurrencyAdapter<Balances, ()>`?

Reply:
Well, I understand that. Its unfortunate that `correct_and_deposit_fee` does not return the final fee like EVMCurrencyAdapter does. Would have loved to reuse the `pallet_transaction_payment::CurrencyAdapter<Balances, ()>`. Something we can definately improve on the upstream so that we dont have to maintain this implemntation per runtime.

Reply:
Okay, with a closer look, there is indeed a hook we can use, namely `OnUnbalanced`.